### PR TITLE
Add GitHub bug, issue, PR templates

### PR DESCRIPTION
**Pull request by andrii-i**
Adds GitHub bug, issue, PR templates for better user, developer, and contributor experience on GitHub.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+title: ''
+labels: 'enhancement'
+assignees: ''
+---
+
+<!-- Welcome! Thank you for contributing. These HTML comments will not render in the issue, but you can delete them once you've read them if you prefer! -->
+
+<!--
+Thanks for thinking of a way to improve Jupyter Scheduler. If this solves a problem for you, then it probably solves that problem for lots of people! So the whole community will benefit from this request.
+Before creating a new feature request please search the issues for relevant feature requests.
```
Comment (srdas) [inline .github/ISSUE_TEMPLATE/feature_request.md:13]: Amend this to: "Before creating a new feature request please search the extant issues for relevant feature requests. See: https://github.com/jupyter-server/jupyter-scheduler/issues."

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<!--
+Thanks for contributing to Jupyter Scheduler!
+Please fill out the following items to submit a pull request.
+Please refer to our contributor's guide for more information on installation and usage:
+https://jupyter-scheduler.readthedocs.io/en/latest/contributors/index.html
+-->
+
+## References
+
+<!-- Note issue numbers this pull request addresses (should be at least one, see contributing guidelines above). -->
```
Comment (srdas) [inline .github/pull_request_template.md:10]: Change "Note issue numbers  ... " to "Note the issue numbers ...". Also insert a "the" before contributing guidelines.

**Review by srdas — COMMENTED** [review]
Made a few suggestions, feel free to disregard as needed.

**Review by srdas — APPROVED** [review]
Thanks for the changes, approved.

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+-->
+
+### Problem
+
+<!-- Provide a clear and concise description of what problem this feature will solve. For example:
+* I'm always frustrated when [...] because [...]
+* I would like it if [...] happened when I [...] because [...]
+-->
+
+### Proposed Solution
```
Comment (JasonWeill) [inline .github/ISSUE_TEMPLATE/feature_request.md:24]:
```suggestion
### Proposed solution
```

**Review by JasonWeill — APPROVED** [review]
Just a couple of non-blocking changes, otherwise looking good. Thanks!

**Pull request merged**